### PR TITLE
Add sql logger

### PR DESCRIPTION
**Pull request by danduk82**
This PR adds a `logging.Handler` that can be used to send python logs to an SQL database. Currently only `sqlite3` and `postgresql_psycopg2` are supported.

Here below I just copy the contents of `c2cwsgiutils/sqlalchemylogger/README.md` for further info:

--------- CONTENTS OF `c2cwsgiutils/sqlalchemylogger/README.md` ---------
This module is used to ship logging records to an SQL database.

Currently only `sqlite` and `postgres_psycopg2` are fully supported.

To add the logger in a pyramid ini file use something like:

```
[handlers]
keys = sqlalchemy_logger

[handler_sqlalchemy_logger]
class = sqlalchemy_logger.handlers.SQLAlchemyHandler
#args = ({'url':'sqlite:///logger_db.sqlite3','tablename':'test'},'curl')
args = ({'url':'postgresql://postgres:password@localhost:5432/test','tablename':'test','tableargs': {'schema':'xyz'}},'curl')
level = NOTSET
formatter = generic
propagate = 0
```

if the credentials given in `args = ` section are sufficient, the handler will
create the DB, schema and table it needs directly.

In the above example the second parameter provided `'curl'` is a negative 
filter (any valid regex will work) to avoid writing the matching logs to the 
DB. Useful to filter out health-check specific `User-Agent` headers or so.

To use the handler in a script, you might:

```python
import logging
import time
from sqlalchemylogger.handlers import SQLAlchemyHandler


if __name__ == '__main__':
    logging.basicConfig(
        format='%(asctime)s : %(name)s : %(levelname)s : %(message)s',
        level=logging.DEBUG,
    )
    logger = logging.getLogger(__name__)
    logger_db_engine = {'url':'sqlite:///logger_db.sqlite3'}
    
    logger.addHandler(SQLAlchemyHandler(logger_db_engine))
    logger.info('bla')
    # wait a few seconds because the second thread will write the
    # logs after a timeout
    time.sleep(2)
```
--------- EOF ---------

**NB**
I had to put a couple of `# type: ignore` in `c2cwsgiutils/sqlalchemylogger/models.py` because to have a full support of sqlalchemy types, we should use [sqlalchemy-stubs](https://github.com/dropbox/sqlalchemy-stubs), but unfortunately why I tried to use this module, many other files in the repository raised `mypy` issues.

Full support of  [sqlalchemy-stubs](https://github.com/dropbox/sqlalchemy-stubs) might be added in another PR but would need extensive testing also of many other things.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,4 @@ __pycache__
 /.idea/sonarlint
 /docs/*/build/
 /.mypy_cache
+*.swp
```
Comment (sbrunner) [inline .gitignore]: Should be in the `.gitignore` of your home directory...

File: c2cwsgiutils/sqlalchemylogger/handlers.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+from sqlalchemy.exc import OperationalError, InvalidRequestError
+from sqlalchemy_utils import database_exists, create_database
+import threading
+import time
+import queue
+from .filters import ContainsExpression, DoesNotContainExpression
+from typing import Any, List, Dict
+
+
+module_logs = logging.getLogger(__name__)
```
Comment (sbrunner) [inline c2cwsgiutils/sqlalchemylogger/handlers.py]: As in other places:
```suggestion
LOG = logging.getLogger(__name__)
```

Comment (danduk82) [reply c2cwsgiutils/sqlalchemylogger/handlers.py]: fixed in [this commit](https://github.com/camptocamp/c2cwsgiutils/pull/456/commits/d69b4afe0a58982e8b86aa297af62b9587cea19b)

File: .gitignore (hunk 1 above)
Comment (danduk82) [reply .gitignore]: fixed in [this commit](https://github.com/camptocamp/c2cwsgiutils/pull/456/commits/d69b4afe0a58982e8b86aa297af62b9587cea19b)

**Comment by danduk82** [comment]
@sbrunner thanks for the review, I have moved the `*.swp` section of `.gitignore` to my own one as you suggested.

**Comment by sbrunner** [comment]
Thanks :-)

**Pull request merged**